.pfnci/linux/run.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+USAGE="
+${0} TARGET STAGE [STAGE ...]
+
+Arguments:
+
+- TARGET: Name of the test target. Targets are defined in 'tests' directory.
+- STAGE: Test stage(s) to execute. Possible stages are:
+  - build: Build a docker image used for testing.
+  - push: Push the built docker image so that further test runs can reuse
+          the image.
+  - cache_get: Pull cache from GCS to CACHE_DIR.
+  - cache_put: Push cache from CACHE_DIR to GCS.
+  - test: Run tests.
+
+Environment variables:
+
+- GPU: Number of GPUs available for testing.
+- CACHE_DIR: Directory to store cache files.
+- DOCKER_IMAGE: Base name of the Docker image (without a tag).
+"
+
+set -eu
+
+
+################################################################################
+# Main function
+################################################################################
+
+main() {
+  if (( $# < 2 )); then
+    echo "${USAGE}"
+    exit 1;
+  fi
+
+  TARGET="$1"; shift
+  STAGES="$@"
+
+  repo_root="$(cd "$(dirname "${BASH_SOURCE}")/../.."; pwd)"
+  base_branch="$(cat "${repo_root}/.pfnci/BRANCH")"
```

Comment:
`BRANCH` file will be used to know the base branch of pull-requests.
By adding this file we no longer have to maintain the list of known branches like this:
https://github.com/cupy/cupy/blob/v10.0.0a2/.pfnci/script.sh#L154-L175

When backporting the content of the file should be changed to `v9`.